test/integration/master_routes_test.go

```diff
@@ -171,6 +175,124 @@ func TestWellKnownOAuthOff(t *testing.T) {
 	}
 }
 
+func TestApiGroups(t *testing.T) {
+	testutil.RequireEtcd(t)
```

Comment:
@guangxuli mind if all this refactoring will be a follow up? don't want to increase the diff size of already big change :-)